source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1637 @@
+  ⋁_ = join-of L
+
+ open Joins _≤_ 
+ open is-small-basis h
+
+\end{code}
+
+We now define an auxilary subset/total space which we will use to define the
+upcoming monotone operator. This subset is in some sense a generalized downset
+that depends on ϕ.
+
+\begin{code}
+
+ _↓_ : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → (a : ⟨ L ⟩) → 𝓤 ⊔ 𝓦 ⊔ 𝓥  ̇
+ ϕ ↓ a = Σ b ꞉ B , (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ a) holds) holds
+
+ ↓-to-base : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → (a : ⟨ L ⟩) → ϕ ↓ a → B
+ ↓-to-base ϕ a = pr₁
+
+ ↓-monotonicity-lemma : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
+                      → (x y : ⟨ L ⟩)
+                      → (x ≤ y) holds
+                      → ϕ ↓ x
+                      → ϕ ↓ y
+ ↓-monotonicity-lemma ϕ x y o (b , c) = (b , inclusion c)
+  where
+   inclusion : (Ǝ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ x) holds)) holds
+             → (Ǝ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ y) holds)) holds
+   inclusion = ∥∥-functor untrunc-inclusion
+    where
+     untrunc-inclusion : Σ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ x) holds)
+                       → Σ a' ꞉ ⟨ L ⟩ , ((b , a') ∈ ϕ) × ((a' ≤ y) holds)
+     untrunc-inclusion (a' , p , r) = (a' , p , transitivity-of L a' x y r o)
+
+ ↓-has-sup-implies-monotone : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
+                            → (x y s s' : ⟨ L ⟩)
+                            → (x ≤ y) holds
+                            → (s is-lub-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+                            → (s' is-lub-of (ϕ ↓ y , β ∘ ↓-to-base ϕ y)) holds
+                            → (s ≤ s') holds
+ ↓-has-sup-implies-monotone
+  ϕ x y s s' o (is-upbnd , is-least-upbnd) (is-upbnd' , is-least-upbnd') =
+   is-least-upbnd (s' , s'-is-upbnd)
+  where
+   s'-is-upbnd : (s' is-an-upper-bound-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+   s'-is-upbnd (b , e) = is-upbnd' (↓-monotonicity-lemma ϕ x y o ((b , e)))
+        
+ is-local : (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺)  ̇
+ is-local ϕ = (a : ⟨ L ⟩) → (ϕ ↓ a) is 𝓥 small
+
+ module _ (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)) (i : is-local ϕ) where
+
+  private
+   S' : (a : ⟨ L ⟩) → 𝓥  ̇
+   S' a = resized (ϕ ↓ a) (i a)
+
+   S'-equiv-↓ : (a : ⟨ L ⟩) → S' a ≃ ϕ ↓ a
+   S'-equiv-↓ a  = resizing-condition (i a)
+
+   S'-to-↓ : (a : ⟨ L ⟩) → S' a → ϕ ↓ a
+   S'-to-↓ a = ⌜ S'-equiv-↓ a ⌝
+
+   ↓-to-S' : (a : ⟨ L ⟩) → ϕ ↓ a → S' a 
+   ↓-to-S' a = ⌜ S'-equiv-↓ a ⌝⁻¹
+
+   S'-monotone-ish : (x y : ⟨ L ⟩)
+                   → (x ≤ y) holds
+                   → S' x
+                   → S' y
+   S'-monotone-ish x y o =
+    ↓-to-S' y ∘ ↓-monotonicity-lemma ϕ x y o ∘ S'-to-↓ x
+
+  Γ : ⟨ L ⟩ → ⟨ L ⟩
+  Γ a = ⋁ (S' a , β ∘ pr₁ ∘ S'-to-↓ a)
+
+\end{code}
+
+We show that Γ is monotone.
+
+\begin{code}
+
+  Γ-is-monotone : is-monotone-endomap L Γ
+  Γ-is-monotone x y o =
+   ↓-has-sup-implies-monotone ϕ x y (Γ x) (Γ y) o Γx-is-lub Γy-is-lub
+   where
+    Γx-is-lub : (Γ x is-lub-of (ϕ ↓ x , β ∘ ↓-to-base ϕ x)) holds
+    Γx-is-lub = sup-of-small-fam-is-lub L (β ∘ ↓-to-base ϕ x) (i x)      
```

Comment:
There are lots of trailing whitespaces in the file, such as on this line. Could you remove these? In Emacs, running `delete-trailing-whitespace` will do the job.